### PR TITLE
test: add test for loading session recording snapshots using personal key

### PR DESCRIPTION
**Pull request by marandaneto**
## Problem

Relates to https://posthog.com/questions/getting-snapshots-through-api

I'm not sure this API should be public.
The [docs](https://posthog.com/docs/api/session-recordings#get-api-projects-project_id-session_recordings-id-snapshots) don't mention the required scope but all the other endpoints do specify the scope, assuming `session_recording:read`

Example:
```bash
curl \
-H "Authorization: Bearer $POSTHOG_PERSONAL_API_KEY" \
https://app.posthog.com/api/projects/$projectId/session_recordings/$sessionId/snapshots
```

Throws error:
```json
{"type":"authentication_error","code":"permission_denied","detail":"This action does not support Personal API Key access","attr":null}
```

The `POSTHOG_PERSONAL_API_KEY` used is a org/project owner with full scope.

[API code path](https://github.com/PostHog/posthog/blob/688cc36d35b64802373ce38329e493c445babe5a/posthog/session_recordings/session_recording_api.py#L300-L457).

Similar to this problem https://github.com/PostHog/posthog/pull/20983

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Comment by marandaneto** [comment]
@xvello Since Paul/Ben are OOO, do you have any ideas if this API should have been public and its a bug in our permission system, or is it intended to be private anyways?

**Comment by pauldambra** [comment]
🤔 well, I think this probably should work - but...

I don't think we've done the thinking about what makes it safe (caching/ratelimiting/etags etc).

So, no reason not to allow this - although interested why the customer wants it. But we should follow up with API docs / caching all the good stuff

**Comment by marandaneto** [comment]
> 🤔 well, I think this probably should work (as in we should allow it) - but...
> 
> I don't think we've done the thinking about what makes it safe (caching/ratelimiting/etags etc).
> 
> So, no reason not to allow this - although interested why the customer wants it. But we should follow up with API docs / caching all the good stuff

Gotcha, The API is already documented, the issue is that it does not mention the required scope, this is the first issue.
The second issue is that even with full scope, the API returns 403.
The caching/ratelimiting/etags bits are probably something to follow up on after figuring out the access :sweat_smile:

File: posthog/api/test/test_personal_api_keys.py [hunk 1]
```diff
@@ -440,6 +442,23 @@ def test_works_with_routes_missing_action(self):
         )
         assert response.status_code == status.HTTP_200_OK
 
+    def test_works_with_session_recording_snapshots(self):
```
Comment (benjackwhite) [inline posthog/api/test/test_personal_api_keys.py:445]: I don't think this test should be here. This is about generally testing the API key functionality. If you just want to add a scope and test that it works then it should be part of the api tests for that endpoint (or just not bother as we know generally that adding the `required_scopes` property works.

File: posthog/api/test/test_personal_api_keys.py [hunk 2]
```diff
@@ -440,6 +442,23 @@ def test_works_with_routes_missing_action(self):
...
+            session_id=session_id,
+            deleted=False,
+            storage_version="2023-08-01",
+            object_storage_path="an lts stored object path",
+        )
+
+        self.key.scopes = ["session_recording:read"]
+        self.key.save()
+        response = self.client.patch(
+            f"/api/projects/{self.team.id}/session_recordings/{session_id}/snapshots?version=2"
```
Comment (benjackwhite) [inline posthog/api/test/test_personal_api_keys.py:458]: Generally as to whether this makes sense - sure - "I guess so". The reason we were cagey about it before is this API has changed _a lot_ so we don't want to make it public unless we are really happy to support the format basically forever.

**Review by benjackwhite — CHANGES_REQUESTED** [review]

File: posthog/api/test/test_personal_api_keys.py (hunk 2 above)
Comment (marandaneto) [reply posthog/api/test/test_personal_api_keys.py:458]: so this API should not be public hence even full token permission does not work?
What I am looking after is: hey this should work but we have a permission bug or hey this is not a public API (I can fix the docs if that's the case) and follow up asking their use case to call the API directly.

**Comment by marandaneto** [comment]
This API wasn't meant to be public for now hence adding the test won't make sense.

**Comment by marandaneto** [comment]
issue tracking this request https://github.com/PostHog/posthog/issues/22520

**Pull request closed without merging**